cloudera/datadog_checks/cloudera/api_client_v7.py

```diff
@@ -0,0 +1,112 @@
+import cm_client
+
+from datadog_checks.cloudera.api_client import ApiClient
+from datadog_checks.cloudera.entity_status import ENTITY_STATUS
+from datadog_checks.cloudera.metrics import METRICS
+
+from .common import CLUSTER_HEALTH, HOST_HEALTH
+from datadog_checks.base import AgentCheck
+
+# TODO: Refactor this to be cleaner
+class ApiClientV7(ApiClient):
+    def __init__(self, check, api_client):
+        super(ApiClientV7, self).__init__(check, api_client)
+
+    def collect_data(self):
+        self._collect_clusters()
+
+    def _collect_clusters(self):
+        clusters_resource_api = cm_client.ClustersResourceApi(self._api_client)
+        read_clusters_response = clusters_resource_api.read_clusters(cluster_type='any', view='full')
+        for cluster in read_clusters_response.items:
+            self._collect_cluster_metrics_and_service_check(cluster)
+
+    def _collect_cluster_metrics_and_service_check(self, cluster):
+        self._log.debug('cluster: %s', cluster)
+        cluster_entity_status = ENTITY_STATUS[cluster.entity_status]
+        cluster_name = cluster.name
+        cluster_tags = cluster.tags
+        self._log.debug('cluster_entity_status: %s', cluster_entity_status)
+        self._log.debug('cluster_name: %s', cluster_name)
+        self._log.debug('cluster_tags: %s', cluster_tags)
+
+        if cluster_name:
+            tags = [f'cloudera_cluster:{cluster_name}']
+
+        if cluster_tags:
+            for cluster_tag in cluster_tags:
+                tags.append(f"{cluster_tag.name}:{cluster_tag.value}")
+        
+        message = cluster.entity_status if cluster_entity_status != AgentCheck.OK else None
+        self._check.service_check(CLUSTER_HEALTH, cluster_entity_status, tags=tags, message=message)
+
+        if cluster_name:
+            self._collect_cluster_metrics(cluster_name, tags)
+            self._collect_hosts_metrics_and_service_check(cluster_name)
+
+    def _collect_cluster_metrics(self, cluster_name, tags):
+        time_series_resource_api = cm_client.TimeSeriesResourceApi(self._api_client)
+        metric_names = ','.join(f'last({metric})' for metric in METRICS['cluster'])
+        query = f'SELECT {metric_names} WHERE clusterName="{cluster_name}" AND category=CLUSTER'
+        self._log.debug('query: %s', query)
+        query_time_series_response = time_series_resource_api.query_time_series(query=query)
+        self._collect_query_time_series(query_time_series_response, 'cluster')
+
+    def _collect_hosts_metrics_and_service_check(self, cluster_name):
+        clusters_resource_api = cm_client.ClustersResourceApi(self._api_client)
+        list_hosts_response = clusters_resource_api.list_hosts(cluster_name, view='full')
```

Comment:
Missing (in general) more debug messages showing variable values. They are very useful in support troubleshooting

Reply:
I didn't include the actual responses in the debug logs, since I thought they would be very noisy considering some queries may be hundreds of lines long.